File: apps/backend/app/schemas/inventory_item.py
### backend/app/schemas/inventory_item.py ###
from uuid import UUID
from datetime import datetime
from decimal import Decimal
from pydantic import BaseModel, ConfigDict, Field
from typing import List
from app.schemas.product_category import ProductCategoryBasic

class InventoryItemBase(BaseModel):
    sku: str
    name: str
    price: Decimal
    category_ids: list[UUID] = Field(default_factory=list)

class InventoryItemCreate(InventoryItemBase):
    pass

class InventoryItemRead(InventoryItemBase):
    id: UUID
    company_id: UUID
    created_at: datetime
    categories: List[ProductCategoryBasic] = Field(default_factory=list)
    model_config = ConfigDict(from_attributes=True)

class PaginatedInventoryItems(BaseModel):
    total: int
    skip: int
    limit: int
    items: List[InventoryItemRead]

    model_config = ConfigDict()

class InventoryItemBasic(BaseModel):
    id: UUID
    name: str
    sku: str
    model_config = ConfigDict(from_attributes=True)

File: apps/portal-company/src/types/inventoryItem.ts
import type { ProductCategoryBasic } from './productCategory';

/** Resposta paginada de itens de inventário. */
export interface PaginatedInventoryItems {
  total: number;
  skip: number;
  limit: number;
  items: InventoryItemRead[];
}

/** Campos compartilhados para criar/atualizar um item de inventário. */
export interface InventoryItemBase {
  sku: string;
  name: string;
  price: number;          // decimal
  category_ids: string[]; // UUIDs de categorias
}

/** Payloads */
export type InventoryItemCreate = InventoryItemBase;
export type InventoryItemUpdate = InventoryItemBase;

/** Item completo retornado pela API. */
export interface InventoryItemRead extends InventoryItemBase {
  id: string;
  company_id: string;
  created_at: string;
  /** vindo do backend agora */
  categories?: ProductCategoryBasic[];
}
